NanoAOD: backport to 10_6_X branch of muon pT > 15 GeV passthrough, addition of isotrack charge and muon isStandalone flag.

mdunser opened the pull request:
#### PR description:

backport to 10_6_X branch of muon pT > 15 GeV passthrough, regardless of ID

this is the backport of PR https://github.com/cms-sw/cmssw/pull/32103
additional backports of #33208 and #33212

mdunser committed:
backport to 10_6_X branch of muon pT > 15 GeV passthrough, regardless of ID

cmsbuild commented:
A new Pull Request was created by  @mdunser  for CMSSW_10_6_X.

It involves the following packages:

PhysicsTools/NanoAOD


@cmsbuild, @santocch, @mariadalfonso, @gouskos, @fgolf can you please review it and eventually sign? Thanks.
@gpetruc, @peruzzim, @swertz this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mariadalfonso commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b5f2966884ce2eab6ad959168b1d1754f15eeff1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/10651/summary.html
CMSSW: CMSSW_10_6_X_2020-11-11-1100
SCRAM_ARCH: slc7_amd64_gcc700

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/10651/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 71 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 3214708
 - DQMHistoTests: Total failures: 115
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3214259
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 140 log files, 14 edm output root files, 34 DQM output files

silviodonato commented:
backport of #32103

mariadalfonso commented:
@mdunser 

please update this backport based on the review in master, 
note that we need to safeguard the current ongoing v8 production in 10_6_X
`(run2_nanoAOD_106Xv1 & ~run2_nanoAOD_devel).toModify(xyzTable , xyxVariable = None)`

mdunser committed:
backport safety for pT passthrough

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mdunser commented:
so i *think* i did the right thing here, but this is the first time, so please have a look and let me know if you find something that is wrong and/or faulty.

mdunser committed:
small bugfix, load the right modifiers at beginning of muon config

mdunser committed:
add isotrack passthrough regardless of the IP variables.

mdunser commented:
i added the isotrack adaptions to this branch. also added the loading of the correct modifiers for the muons, which weren't there before. i hope this is what was needed. let me know if not ...

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
so for muons we have 
from PhysicsTools.NanoAOD.nano_eras_cff import *
https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_X/PhysicsTools/NanoAOD/python/muons_cff.py#L2
the PR probably need a rebase

for isoTrack you can add directly the whole nano_eras_cff

mdunser committed:
bugfix adding modifiers for eras

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mdunser commented:
i hope this fixed it? let me know...

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/13918/summary.html
**COMMIT**: c6de68d3b8f4890143dfd496942511642bd1132a
**CMSSW**: CMSSW_10_6_X_2021-03-31-2300/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/32107/13918/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 84 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 3215541
 - DQMHistoTests: Total failures: 76
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215131
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 143 log files, 29 edm output root files, 35 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_6_X_2021-03-31-2300+3b2788/42008/triggerResults): no differences found

sroychow committed:
adding variables to muons & isotracks

sroychow committed:
adding dqm plots of new variables

sroychow committed:
customize function for increasing gen precision

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

sroychow commented:
The last commits include the backport of #33208 and #33212.

mariadalfonso commented:
the backport looks correct, but we have conflict in PhysicsTools/NanoAOD/python/nanoDQM_cff.py (needed before we start the tests )

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/summary.html
**COMMIT**: 90f564381fc2ac62b7dc2e2aa7316a80fe650d15
**CMSSW**: CMSSW_10_6_X_2021-04-14-1100/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/32107/14243/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)```136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log```
- [136.8523](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2)```136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2/step2_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2.log```
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)```136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL/step2_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)
- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18)
- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5)
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2)
- [136.788](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM)
- [136.85](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM)
- [10042.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017)
- [158.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT)
- [11624.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021)
- [10024.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25)
- [10824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU)
- [20034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17)
- [21234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21)
- [27434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35)
- [29034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/29034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D41+RecoFullGlobal_2023D41+HARVESTFullGlobal_2023D41)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25)
</details>



## RelVals-INPUT

- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/4.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1)```4.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1/step2_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1.log```
- [136.72411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)```136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM/step2_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM.log```
- [136.76111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.76111_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)```136.76111_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM/step2_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)
- [136.77211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.77211_RunJetHT2016H_reminiaodUL+RunJetHT2016H_reminiaodUL+REMINIAOD_data2016UL+HARVESTDR2_REMINIAOD_data2016UL)
- [136.7721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016)
- [136.7722](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X)
- [136.7802](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7802_RunHLTPhy2017B_AODextra+RunHLTPhy2017B_AODextra+DQMHLTonAODextra_2017+HARVESTDQMHLTonAOD_2017)
- [136.7803](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7803_RunHLTPhy2017B_RAWAOD+RunHLTPhy2017B_RAWAOD+DQMHLTonRAWAOD_2017+HARVESTDQMHLTonAOD_2017)
- [136.7801](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7801_RunHLTPhy2017B_AOD+RunHLTPhy2017B_AOD+DQMHLTonAOD_2017+HARVESTDQMHLTonAOD_2017)
- [136.7952](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.7952_RunJetHT2017C_94Xv2NanoAODINPUT+RunJetHT2017C_94Xv2NanoAODINPUT+NANOEDM2017_94XMiniAODv2+HARVESTNANOAOD2017_94XMiniAODv2)
- [136.83111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.83111_RunJetHT2017F_reminiaodUL+RunJetHT2017F_reminiaodUL+REMINIAOD_data2017UL+HARVEST2017_REMINIAOD_data2017UL)
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)
- [136.8521](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.8521_RunJetHT2018A_nano+RunJetHT2018A_nano+NANOEDM2018_102Xv1+HARVESTNANOAOD2018_102Xv1)
- [136.8523](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2)
- [136.8522](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.8522_RunJetHT2018A_nanoUL+RunJetHT2018A_nanoUL+NANOEDM2018_106Xv1+HARVESTNANOAOD2018_106Xv1)
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18)
- [140.54](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/140.54_RunPA2013+RunPA2013+RECO_PPbData+HARVEST_PPbData)
- [140.57](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/140.57_RunHI2018Reduced+RunHI2018Reduced+RECOHID18+HARVESTDHI18)
- [1325.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.5_ProdZEE_13_reminiaodINPUT+ProdZEE_13_reminiaodINPUT+REMINIAOD_mc2016+HARVESTDR2_REMINIAOD_mc2016)
- [1325.51](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.51_TTbar_13_94XreminiaodINPUT+TTbar_13_94XreminiaodINPUT+REMINIAOD_mc2017+HARVESTUP17_REMINIAOD_mc2017)
- [1325.516](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.516_TTbar_13_reminiaod2016UL_preVFP_INPUT+TTbar_13_reminiaod2016UL_preVFP_INPUT+REMINIAOD_mc2016UL_preVFP+HARVESTDR2_REMINIAOD_mc2016UL_preVFP)
- [1325.5161](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.5161_TTbar_13_reminiaod2016UL_postVFP_INPUT+TTbar_13_reminiaod2016UL_postVFP_INPUT+REMINIAOD_mc2016UL_postVFP+HARVESTDR2_REMINIAOD_mc2016UL_postVFP)
- [1325.517](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.517_TTbar_13_reminiaod2017UL_INPUT+TTbar_13_reminiaod2017UL_INPUT+REMINIAOD_mc2017UL+HARVESTUP17_REMINIAOD_mc2017UL)
- [1325.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.6_TTbar_13_94Xv1NanoAODINPUT+TTbar_13_94Xv1NanoAODINPUT+NANOAODMC2017_94XMiniAODv1)
- [1325.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.61_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOAODMC2017_106XMiniAODv1)
- [1325.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.7_TTbar_13_94Xv2NanoAODINPUT+TTbar_13_94Xv2NanoAODINPUT+NANOEDMMC2017_94XMiniAODv2+HARVESTNANOAODMC2017_94XMiniAODv2)
- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)
- [1325.8](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.8_TTbar_13_94Xv1NanoAODINPUT+TTbar_13_94Xv1NanoAODINPUT+NANOEDMMC2017_94XMiniAODv1+HARVESTNANOAODMC2017_94XMiniAODv1)
- [1325.518](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1325.518_TTbar_13_reminiaod2018UL_INPUT+TTbar_13_reminiaod2018UL_INPUT+REMINIAOD_mc2018UL+HARVESTUP18_REMINIAOD_mc2018UL)
- [1329.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1329.1_ZEE_13_80XNanoAODINPUT+ZEE_13_80XNanoAODINPUT+NANOEDMMC2016_80X+HARVESTNANOAODMC2016_80X)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/134.0_MinimumBiasBS+RunMinBias2011B+RECODR1+ALCAPROMPT+ALCAHARVD)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/144.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCAR1)
- [1001.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1001.2_RunZeroBias2017F+RunZeroBias2017F+TIER0EXPRUN2+ALCAEXPRUN2+ALCAHARVDSIPIXELCAL)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5)
- [1003.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1002.0_RRD+RunMinBias2011A+RECODR1+COPYPASTE)
- [1030.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1030.0_RunHLTPhy2017B+RunHLTPhy2017B+TIER0EXPHPBS+ALCASPLITHPBS+ALCAHARVDHPBS+ALCAHARVDHPBSLOWPU)
- [1040.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14243/runTheMatrixINPUT-results/1040.0_RunZeroBias2017F+RunZeroBias2017F+TIER0RAWSIPIXELCAL+ALCASPLITSIPIXELCAL+ALCAHARVDSIPIXELCAL)
</details>

sroychow committed:
add missing era modifier include

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
@mdunser @sroychow 

https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_X/PhysicsTools/NanoAOD/python/nanoDQM_cff.py#L70-L78
so things are not correct, as you can see the two eras are already in the release.

you need to rebase the PR on top of the current 10_6_X !
follow instruction here 
http://cms-sw.github.io/tutorial-resolve-conflicts.html

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented on PhysicsTools/NanoAOD/python/nanoDQM_cff.py:
those two modifiers were already included :)
just two lines above

mdunser committed:
bugfix to remove duplicate imports

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/summary.html
**COMMIT**: 3f0509b3f96b114c99735f836ee8e1e79728d6f3
**CMSSW**: CMSSW_10_6_X_2021-04-16-1100/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/32107/14282/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)```136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log```


## RelVals-INPUT

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrixINPUT-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)```136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log```
- [136.7721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrixINPUT-results/136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016)```136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016/step2_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016.log```
- [136.7722](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrixINPUT-results/136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X)```136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X/step2_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [1325.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrixINPUT-results/1325.5_ProdZEE_13_reminiaodINPUT+ProdZEE_13_reminiaodINPUT+REMINIAOD_mc2016+HARVESTDR2_REMINIAOD_mc2016)
- [1329.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14282/runTheMatrixINPUT-results/1329.1_ZEE_13_80XNanoAODINPUT+ZEE_13_80XNanoAODINPUT+NANOEDMMC2016_80X+HARVESTNANOAODMC2016_80X)
</details>

mariadalfonso commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/summary.html
**COMMIT**: 3f0509b3f96b114c99735f836ee8e1e79728d6f3
**CMSSW**: CMSSW_10_6_X_2021-04-19-2300/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14343/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)```136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log```


## RelVals-INPUT

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrixINPUT-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)```136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log```
- [136.7721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrixINPUT-results/136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016)```136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016/step2_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016.log```
- [136.7722](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrixINPUT-results/136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X)```136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X/step2_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [1325.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrixINPUT-results/1325.5_ProdZEE_13_reminiaodINPUT+ProdZEE_13_reminiaodINPUT+REMINIAOD_mc2016+HARVESTDR2_REMINIAOD_mc2016)
- [1329.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14343/runTheMatrixINPUT-results/1329.1_ZEE_13_80XNanoAODINPUT+ZEE_13_80XNanoAODINPUT+NANOEDMMC2016_80X+HARVESTNANOAODMC2016_80X)
</details>

mariadalfonso commented:
@mdunser @sroychow 
please have a look 

 ```
 File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14343/CMSSW_10_6_X_2021-04-19-2300/python/DQMOffline/Configuration/DQMOffline_cff.py", line 190, in <module>
    from PhysicsTools.NanoAOD.nanoDQM_cff import nanoDQM
  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14343/CMSSW_10_6_X_2021-04-19-2300/python/PhysicsTools/NanoAOD/nanoDQM_cff.py", line 83, in <module>
    _Isotk_nocharge = [plot for plot in nanoDQM.vplots.IsoTrack.plots if plot.name.value() != 'charge']
AttributeError: 'PSet' object has no attribute 'IsoTrack'
```

gouskos commented:
> @mdunser @sroychow
> please have a look
> 
> ```
> File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14343/CMSSW_10_6_X_2021-04-19-2300/python/DQMOffline/Configuration/DQMOffline_cff.py", line 190, in <module>
>    from PhysicsTools.NanoAOD.nanoDQM_cff import nanoDQM
>  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14343/CMSSW_10_6_X_2021-04-19-2300/python/PhysicsTools/NanoAOD/nanoDQM_cff.py", line 83, in <module>
>    _Isotk_nocharge = [plot for plot in nanoDQM.vplots.IsoTrack.plots if plot.name.value() != 'charge']
> AttributeError: 'PSet' object has no attribute 'IsoTrack'
> ```

I think you should do something like here:
https://github.com/cms-sw/cmssw/blob/3f0509b3f96b114c99735f836ee8e1e79728d6f3/PhysicsTools/NanoAOD/python/nanoDQM_cff.py#L70-78

sroychow commented:
Hi @gouskos the lines added for the Isotrack should work in principle since it does the same thing done in line 71 - 75. The problem in my opinion comes from line 67 since for run2_miniAOD_80XLegacy, the IsoTrack Pset is removed. The simplest solution is to move line 67 to the end of the file(this works in the local tests I did).

mdunser committed:
moving a line further down to hopefully fix a bug

cmsbuild commented:
Pull request #32107 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b2788/14539/summary.html
**COMMIT**: eabe84355399d16bb9189429a73d94a9d4020081
**CMSSW**: CMSSW_10_6_X_2021-04-23-1100/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32107/14539/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 158 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 3215541
 - DQMHistoTests: Total failures: 76
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215131
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.329 KiB( 34 files compared)
 - DQMHistoSizes: changed ( 136.8523 ): 0.329 KiB        Physics/NanoAODDQM
 - Checked 143 log files, 29 edm output root files, 35 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_6_X_2021-04-23-1100+3b2788/42378/triggerResults): no differences found

mariadalfonso commented:
+xpog

changes inline with master (3PRs backport)
106Xv1 unchanged

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_6_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_12_0_X is complete. This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
